Compiler bug in for_in methods call

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 21.04
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1) 10.3.0

getwd: /var/www/vproject
vmodules: /home/tanguy/.vmodules
vroot: /var/www/vproject/v
vexe: /var/www/vproject/v/v
vexe mtime: 2022-01-07 04:37:44
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 c1711b8.60527a5

Git version: git version 2.30.2
Git vroot status: weekly.2021.52.1-74-g60527a56
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
Playing with #12621 and #12741

`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/structs/for_in_methods_call.v`
```v
struct Foo {}

fn (f Foo) a() string {
	return 'method_a'
}

fn (f Foo) b() string {
	return 'method_b'
}

fn main() {
	f := Foo{}

	$for method in Foo.methods {
		x := f.$method()
		println(x)
		println(f.$method())
	}
}
```

**What did you expect to see?**
Maybe an error with a message


**What did you see instead?**
```
==================
/tmp/v_1000/for_in_methods_call.6920207971051817627.tmp.c:10963: error: ',' expected (got ";")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/comptime.v
+++ b/vlib/v/gen/c/comptime.v
@@ -150,7 +150,11 @@ fn (mut g Gen) comptime_call(mut node ast.ComptimeCall) {
 				}
 			}
 		}
-		g.write(' ); // vweb action call with args')
+		if g.inside_call {
+			g.write(')')
+		} else {
+			g.write(');')
+		}
 		return
 	}
 	mut j := 0

--- a/vlib/v/tests/comptime_call_in_fn_call_test.v
+++ b/vlib/v/tests/comptime_call_in_fn_call_test.v
@@ -1,0 +1,29 @@
+struct Foo {}
+
+fn (f Foo) a() string {
+	return 'method_a'
+}
+
+fn (f Foo) b() string {
+	return 'method_b'
+}
+
+fn test_comptime_call_in_fn_call() {
+	f := Foo{}
+
+	mut rets := []string{}
+
+	$for method in Foo.methods {
+		x := f.$method()
+		println(x)
+		println(f.$method())
+		rets << get_string(f.$method())
+	}
+	assert rets.len == 2
+	assert rets[0] == 'method_a'
+	assert rets[1] == 'method_b'
+}
+
+fn get_string(s string) string {
+	return s
+}

